libnopegl/src/node_rtt.c

```diff
@@ -155,10 +167,30 @@ static int rtt_init(struct ngl_node *node)
             LOG(ERROR, "render targets cannot have a data source");
             return NGL_ERROR_INVALID_ARG;
         }
+
+        const struct texture_priv *texture_priv = info.texture_priv;
+        struct texture_params *params = &texture_priv->params;
+        if (s->width != params->width || s->height != params->height) {
+            LOG(ERROR, "color and depth texture dimensions do not match: %dx%d != %dx%d",
+                s->width, s->height, params->width, params->height);
+            return NGL_ERROR_INVALID_ARG;
+        }
+
         if (!(gpu_ctx->features & NGLI_FEATURE_DEPTH_STENCIL_RESOLVE) && o->samples > 0) {
             LOG(ERROR, "context does not support resolving depth/stencil attachments");
             return NGL_ERROR_GRAPHICS_UNSUPPORTED;
         }
+
+        params->usage |= NGLI_TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT;
+        s->layout.depth_stencil.format = params->format;
+        s->layout.depth_stencil.resolve = o->samples > 1;
+    } else {
+        int depth_format = NGLI_FORMAT_UNDEFINED;
+        if (s->renderpass_info.features & NGLI_RENDERPASS_FEATURE_STENCIL)
+            depth_format = ngli_gpu_ctx_get_preferred_depth_stencil_format(gpu_ctx);
+        else if (s->renderpass_info.features & NGLI_RENDERPASS_FEATURE_DEPTH)
+            depth_format = ngli_gpu_ctx_get_preferred_depth_format(gpu_ctx);
+        s->layout.depth_stencil.format = depth_format;
```

Comment:
I think this belongs in another commit (this is currently in "rtt: move attachment dimensions check to rtt_init()")

Reply:
Yep. Moved to the next commit.